ch05/07_gpt_to_llama/converting-gpt-to-llama2.ipynb

```diff
@@ -143,7 +143,7 @@
     "- LayerNorm normalizes inputs using mean and variance, while RMSNorm uses only the root mean square, which improves computational efficiency\n",
     "- The RMSNorm operation is as follows, where $x$ is the input $\\gamma$ is a trainable parameter (vector), and $\\epsilon$ is a small constant to avoid zero-division errors:\n",
     "\n",
-    "$$y = \\frac{x}{\\sqrt{\\text{RMS}[x]} + \\epsilon} * \\gamma$$\n",
+    "$$y = \\frac{x}{\\sqrt{\\text{RMS}[x^2]} + \\epsilon} * \\gamma$$\n",
```

Comment:
Thanks for the change here! I would keep it as it is though, otherwise it would look like it deviates from the PyTorch one (https://pytorch.org/docs/stable/generated/torch.nn.RMSNorm.html)

I think it's just a notational convention since the square happens inside as per RMSNorm paper:

<img width="576" alt="Screenshot 2024-09-24 at 6 38 01 PM" src="https://github.com/user-attachments/assets/c995d426-4fc0-4bff-adee-715cdda4ba2d">




It is still weird to have the square root outside though. In my opinion, it should be called 

$$y = \\frac{x}{\\sqrt{\\text{MS}[x]} + \\epsilon} * \\gamma$$

(MS for "Mean square" instead of RMS)


```suggestion
    "$$y_i = \frac{x_i}{\text{RMS}(x)} \gamma_i, \quad \text{where} \quad \text{RMS}(x) = \sqrt{\epsilon + \frac{1}{n} \sum x_i^2}$$\n",
```

Reply:
Yeah, I agree, itz 

> It is still weird to have the square root outside though. In my opinion, it should be called
> 
> y = x MS [ x ] + ϵ ∗ γ
> 
> (MS for "Mean square" instead of RMS)

Yeah, I agree, this is actually more aligned with the standard implementation of RMS normalization

Reply:
FYI I have raised this formula as an issue to the pytorch team: https://github.com/pytorch/pytorch/issues/136597

Reply:
@rasbt About the $\+\hspace{1mm}\epsilon$ in the formula, is this inside or outside the root (in the pytorch formula, it's inside the root, different to your formula where it's outside the root)

Reply:
Good call out. It should be inside the square root. The above is thus not entirely correct but I used the correct formula when updating it in the notebook yesterday:

$$y_i = \frac{x_i}{\text{RMS}(x)} \gamma_i, \quad \text{where} \quad \text{RMS}(x) = \sqrt{\epsilon + \frac{1}{n} \sum x_i^2}$$

Reply:
Yes, thanks!

Reply:
Oh nice!